connector_magento/components/backend_adapter.py

```diff
@@ -0,0 +1,426 @@
+                    arguments.pop()
+            start = datetime.now()
+            try:
+                result = self.api_call(
+                    method, arguments, http_method=http_method, storeview=storeview
+                )
+            except Exception:
+                _logger.error("api.call('%s', %s) failed", method, arguments)
+                raise
+            else:
+                _logger.debug(
+                    "api.call('%s', %s) returned %s in %s seconds",
+                    method,
+                    arguments,
+                    result,
+                    (datetime.now() - start).seconds,
+                )
+            # Uncomment to record requests/responses in ``recorder``
+            # record(method, arguments, result)
+            return result
+        except (socket.gaierror, socket.error, socket.timeout) as err:
+            raise NetworkRetryableError(
+                "A network error caused the failure of the job: " "%s" % err
+            )
+        except xmlrpc.client.ProtocolError as err:
+            if err.errcode in [
+                502,  # Bad gateway
+                503,  # Service unavailable
+                504,
+            ]:  # Gateway timeout
+                raise RetryableJobError(
+                    "A protocol error caused the failure of the job:\n"
+                    "URL: %s\n"
+                    "HTTP/HTTPS headers: %s\n"
+                    "Error code: %d\n"
+                    "Error message: %s\n"
+                    % (err.url, err.headers, err.errcode, err.errmsg)
+                )
+            else:
+                raise
+
+
+class MagentoCRUDAdapter(AbstractComponent):
+    """ External Records Adapter for Magento """
+
+    # pylint: disable=method-required-super
+
+    _name = "magento.crud.adapter"
+    _inherit = ["base.backend.adapter", "base.magento.connector"]
+    _usage = "backend.adapter"
+
+    def search(self, filters=None):
+        """Search records according to some criterias
+        and returns a list of ids"""
+        raise NotImplementedError
+
+    def read(self, external_id, attributes=None, storeview=None):
+        """ Returns the information of a record """
+        raise NotImplementedError
+
+    def search_read(self, filters=None):
+        """Search records according to some criterias
+        and returns their information"""
+        raise NotImplementedError
+
+    def create(self, data):
+        """ Create a record on the external system """
+        raise NotImplementedError
+
+    def write(self, external_id, data):
+        """ Update records on the external system """
+        raise NotImplementedError
+
+    def delete(self, external_id):
+        """ Delete a record on the external system """
+        raise NotImplementedError
+
+    def _call(self, method, arguments=None, http_method=None, storeview=None):
+        try:
+            magento_api = getattr(self.work, "magento_api")  # noqa: B009
```

Comment:
Don's just silence the linter in this case, as it is easy to fix:
```suggestion
            magento_api = self.work["magento_api"]
```